nrf52840-hal/src/ieee802154.rs

```diff
@@ -243,6 +259,50 @@ impl<'c> Radio<'c> {
             .write(|w| w.txpower().variant(power._into()));
     }
 
+    /// Sample the received signal power (i.e. the presence of possibly interfering signals)
+    /// within the bandwidth of the currently used channel for sample_cycles iterations.
+    /// The result can be used to find a suitable ED threshold for Energy Detection-based CCA mechanisms.
+    ///
+    /// Returns the *maximum* measurement recorded during sampling as reported by the hardware (not in dBm!)
+    ///
+    /// For details, see Section 6.20.12.3 Energy detection (ED) of the PS.
+    /// RSSI samples are averaged over a measurement time of 8 symbol periods (128 μs).
+    pub fn energy_detection_scan(&mut self, sample_cycles: u32) -> u8 {
+        self.needs_enable = true;
+
+        // clear related events
+        self.radio.events_edend.reset();
```

Comment:
I would still like to move this statement to just before the `tasks_edstart.write` statement.